src/rules/padding.ts

```diff
@@ -0,0 +1,384 @@
+/**
+ * @fileoverview Rule to require or disallow newlines between jest functions,
+ *   based on eslint/padding-line-between-statements by Toru Nagashima
+ */
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const astUtils = require('eslint/lib/util/ast-utils');
```

Comment:
Can we switch this to using the `import` syntax if possible

Reply:
Turns out we just used three tiny pieces from this ast-utils. I moved them into our `padding.ts` because I was starting to get concerned that ast-utils may not have been a stable thing to depend on.

This is all that was used from ast utils (types added).

```ts
const STATEMENT_LIST_PARENTS = new Set([
  'Program',
  'BlockStatement',
  'SwitchCase',
  'SwitchStatement',
]);

const isTokenOnSameLine = (
  left: Node | AST.Token,
  right: Node | AST.Token,
): boolean => left.loc.end.line === right.loc.start.line;

const isSemicolonToken = (token: AST.Token): boolean =>
  token.value === ';' && token.type === 'Punctuator';
```